### PR TITLE
Fix keyboard covering up chat messages

### PR DESCRIPTION
**Pull request by Chiamaka**
This PR aims to solve this issue https://github.com/zulip/zulip-mobile/issues/180

Please let me know your suggestions.

Thanks

### CODE REVIEW

File: src/common/ZulipStatusBar.js [hunk 1]
```diff
@@ -31,11 +31,11 @@ class ZulipStatusBar extends PureComponent {
...
     const style = { height: hidden ? 0 : STATUSBAR_HEIGHT, backgroundColor };
     const statusBarStyle = !barStyle
-     ? getStatusBarStyle(backgroundColor, textColor, theme)
-     : barStyle;
+      ? getStatusBarStyle(backgroundColor, textColor, theme)
+      : barStyle;
     const statusBarColor = getStatusBarColor(backgroundColor, theme);
     return (
-      <View style={style}>
+      <View style={[style, { zIndex: 1 }]}>
```
Comment (borisyankov) [inline src/common/ZulipStatusBar.js]: Extract the `{ zIndex: 1 }` style too.

File: src/nav/MainNavBar.js [hunk 2]
```diff
@@ -49,7 +49,7 @@ class MainNavBar extends PureComponent {
     const leftPress = editMessage ? actions.cancelEditMessage : onPressStreams;
 
     return (
-      <View style={[styles.navBar, { backgroundColor }]}>
+      <View style={[styles.navBar, { backgroundColor, zIndex: 1 }]}>
```
Comment (borisyankov) [inline src/nav/MainNavBar.js]: And here too.
We have to pass an object of `{ backgroundColor }` as it is a dynamic value.

**Comment by Chiamaka** [comment]
okay, on this right now. I'll also test on Android and report what I find. Thanks for the feedback

**Comment by borisyankov** [comment]
I just noticed that `eslint` is also complaining about what I said :)
Make sure to run `jest` `npm run test:lint` and `flow` locally, to verify that all tests work correctly, and to get early hints on something that might be wrong.

File: src/chat/Chat.js [hunk 3]
```diff
@@ -55,7 +55,7 @@ export default class Chat extends PureComponent {
     const WrapperView = Platform.OS === 'ios' ? KeyboardAvoidingView : View;
 
     return (
-      <WrapperView style={styles.screen} behavior="padding">
+      <WrapperView style={styles.screen} behavior="position" contentContainerStyle={{ flex: 1 }}>
```
[the remark this answers is not in the document]
Comment (Chiamaka) [reply src/chat/Chat.js]: I have a question about this. Please look at line 53, I am confused about the `this.context`, can you help me out? Do I define another styles object?

[the remark this answers is not in the document]
Comment (vishwesh3) [reply src/chat/Chat.js]: Yes, you can define another `styles` object. Somewhat similar to [Popup.js](https://github.com/zulip/zulip-mobile/blob/master/src/common/Popup.js) where we have context styles as well as local styles.

context styles are from [StyleProvider.js](https://github.com/zulip/zulip-mobile/blob/master/src/boot/StylesProvider.js) which are theme specific.

**Comment by Chiamaka** [comment]
Update: It wasn't compatible with Android so after a few tweaks I have this working on Android. Not perfect because the `MainNavBar` disappears but still working on it

**Comment by Chiamaka** [comment]
I've updated the PR. Please check @borisyankov. Thanks

**Comment by borisyankov** [comment]
I think this is a tricky task @Chiamaka so don't get discouraged that it will likely need lots of iterations.

**Comment by Chiamaka** [comment]
It's okay @borisyankov. It's all part of the learning process :)

**Comment by borisyankov** [comment]
Hey, @Chiamaka are you still working on this?

**Comment by Chiamaka** [comment]
Yeah sure!

**Comment by Chiamaka** [comment]
Closed in favor of https://github.com/zulip/zulip-mobile/pull/1364

**Pull request closed without merging**